pkg/asset/cluster/aws/aws.go

```diff
@@ -29,48 +30,100 @@ func Metadata(clusterID, infraID string, config *types.InstallConfig) *awstypes.
 // PreTerraform performs any infrastructure initialization which must
 // happen before Terraform creates the remaining infrastructure.
 func PreTerraform(ctx context.Context, clusterID string, installConfig *installconfig.InstallConfig) error {
-	if len(installConfig.Config.Platform.AWS.Subnets) == 0 {
-		return nil
+	if len(installConfig.Config.Platform.AWS.Subnets) != 0 {
```

Comment:
Rather than have a big if block, let's have `PreTerraform` call two functions: One function will tag the subnets, and the other function will tag the iam roles. The subnet-tagging function can return early when there are no user-provided subnets.